halo2_proofs/src/plonk/vanishing/prover.rs

```diff
@@ -121,7 +127,9 @@ impl<C: CurveAffine> Constructed<C> {
             .rev()
             .fold(domain.empty_coeff(), |acc, eval| acc * xn + eval);
 
+        #[cfg(feature = "zero-knowledge")]
         let random_eval = eval_polynomial(&self.committed.random_poly, *x);
+        #[cfg(feature = "zero-knowledge")]
```

Comment:
is it better to use brace to put them together? like

```
#[cfg(feature = "zero-knowledge")]
{
  let random_eval = eval_polynomial(&self.committed.random_poly, *x);
  transcript.write_scalar(random_eval)?;
}
```

more easy to read code

```